### PR TITLE
Replaced for with foreach and replace if ,else if, else  condition with match

### PR DESCRIPTION
**Pull request by akashsethi24**
In some Places for the loop was used but it can be replaced by the for each.
In One file if else if else was used so I replaced the same with match.

### CODE REVIEW

File: clients/src/main/java/org/apache/kafka/common/network/ByteBufferReceive.java [hunk 1]
```diff
@@ -33,8 +33,7 @@ public ByteBufferReceive(String source, ByteBuffer... buffers) {
         super();
         this.source = source;
         this.buffers = buffers;
-        for (int i = 0; i < buffers.length; i++)
-            remaining += buffers[i].remaining();
+        for (ByteBuffer buffer : buffers) remaining += buffer.remaining();
```
Comment (guozhangwang) [inline clients/src/main/java/org/apache/kafka/common/network/ByteBufferReceive.java]: Better be two lines for debugging.

File: streams/src/test/java/org/apache/kafka/streams/kstream/internals/KTableKTableJoinTest.java [hunk 2]
```diff
@@ -138,8 +138,8 @@ public void testJoin() throws Exception {
 
         // push all four items to the primary stream. this should produce four items.
 
-        for (int i = 0; i < expectedKeys.length; i++) {
-            driver.process(topic1, expectedKeys[i], "X" + expectedKeys[i]);
+        for (int expectedKey1 : expectedKeys) {
```
Comment (guozhangwang) [inline streams/src/test/java/org/apache/kafka/streams/kstream/internals/KTableKTableJoinTest.java]: Does this have to be a different variable name than line 131 above? Same below.

**Review by guozhangwang — COMMENTED** [review]
Minor comments. Otherwise LGTM.

File: streams/src/test/java/org/apache/kafka/streams/kstream/internals/KTableKTableJoinTest.java (hunk 2 above)
Comment (akashsethi24) [reply streams/src/test/java/org/apache/kafka/streams/kstream/internals/KTableKTableJoinTest.java]: Thank you for pointing out.
 just updated.
Let me know anything else needs to be fixed.

File: core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala [hunk 3]
```diff
@@ -306,7 +306,8 @@ object ConsumerGroupCommand extends Logging {
       val offsetFetchResponse = OffsetFetchResponse.readFrom(channel.receive().payload())
 
       offsetFetchResponse.requestInfo.foreach { case (topicAndPartition, offsetAndMetadata) =>
-        if (offsetAndMetadata == OffsetMetadataAndError.NoOffset) {
+        offsetAndMetadata match {
+          case OffsetMetadataAndError.NoOffset => {
```
Comment (guozhangwang) [inline core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala]: nit: indentation. Same below.

File: streams/src/test/java/org/apache/kafka/streams/kstream/internals/KTableKTableJoinTest.java (hunk 2 above)
Comment (guozhangwang) [reply streams/src/test/java/org/apache/kafka/streams/kstream/internals/KTableKTableJoinTest.java]: This line (314) could be renamed to `expected Key` as well.

**Comment by guozhangwang** [comment]
Merged to trunk and 0.10.2.

**Pull request closed without merging**